store/db/memdb.go

```diff
@@ -0,0 +1,462 @@
+package db
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/google/btree"
```

Comment:
would use tidwall (https://github.com/tidwall/btree)

Reply:
Yes, @cool-develope please use the generic version of tidwall